src/colorizer/Collection.ts

```diff
@@ -235,42 +243,70 @@ export default class Collection {
    * Asynchronously loads a Collection object from the provided URL.
    * @param collectionParam The URL of the resource. This can either be a direct path to
    * collection JSON file or the path of a directory containing `collection.json`.
-   * @param fetchMethod Optional. The fetch command used to retrieve the URL.
+   * @param options Optional configuration, containing any of the following properties:
+   * - `fetchMethod` optional override for the fetch method, used to retrieve the URL.
+   * - `reportWarning` optional callback for reporting warning messages during loading.
    * @throws Error if the JSON could not be retrieved or is an unrecognized format.
    * @returns A new Collection object containing the retrieved data.
    */
-  public static async loadCollection(collectionParam: string, fetchMethod = fetchWithTimeout): Promise<Collection> {
+  public static async loadCollection(
+    collectionParam: string,
+    options: Partial<{ fetchMethod: typeof fetchWithTimeout; reportWarning: ReportWarningCallback }> = {}
+  ): Promise<Collection> {
     const absoluteCollectionUrl = Collection.formatAbsoluteCollectionPath(collectionParam);
 
     let response;
     try {
+      const fetchMethod = options.fetchMethod ?? fetchWithTimeout;
       response = await fetchMethod(absoluteCollectionUrl, DEFAULT_FETCH_TIMEOUT_MS);
     } catch (e) {
-      throw new Error(`Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': '${e}'`);
+      throw new Error(
+        LoadErrorMessage.UNREACHABLE_COLLECTION +
+          " This may be due to a network issue, the server being unreachable, or a misconfigured URL." +
+          " Please check your network access."
+      );
     }
     if (!response.ok) {
-      throw new Error(`Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': Fetch failed.`);
+      console.error(`Failed to fetch collections JSON from url '${absoluteCollectionUrl}':`, response);
```

Comment:
just curious: why are you now also logging this error?

Reply:
Also, what's the situations when the fetch doesn't return an error but the response isn't `ok`?

Reply:
Any type of server error like a 4XX or 5XX error would return a non-`ok` response.

I think you're right to point this out and I can probably discard it; I think this was a holdover from before I changed the Load dialog to display error messages from both the collection and the dataset (and it used to only show the Dataset error)